backends/dpdk/dpdkArch.cpp

```diff
@@ -1526,6 +1526,8 @@ const IR::Node* SplitP4TableCommon::postorder(IR::IfStatement* statement) {
 
 const IR::Node* SplitP4TableCommon::postorder(IR::SwitchStatement* statement) {
     auto expr = statement->expression;
+    if (!expr->is<IR::Member>())
```

Comment:
could swap these or check if member is null below.
There is also a set of utilities in https://github.com/p4lang/p4c/blob/main/frontends/p4/tableApply.h that make it easier to detect this pattern.